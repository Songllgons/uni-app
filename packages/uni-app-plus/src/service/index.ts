import * as uni from './api'
import { UniServiceJSBridge } from './bridge'
import { registerApp as __registerApp, getApp } from './framework/app'
import {
  definePage as __definePage,
  registerPage as __registerPage,
  getCurrentPages,
} from './framework/page'

export default {
  uni,
  getApp,
  getCurrentPages,
  __definePage,
  __registerApp,
  __registerPage,
  UniServiceJSBridge,
}
