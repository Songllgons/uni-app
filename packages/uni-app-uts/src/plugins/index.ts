import { init as initAndroidPlugins } from './android'
import { init as initIOSPlugins } from './ios'

export function initAndroid() {
  import('./errorReporting')
  return initAndroidPlugins()
}

export function initIOS() {
  import('./errorReporting')
  return initIOSPlugins()
}
