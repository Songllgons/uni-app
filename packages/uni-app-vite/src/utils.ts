import path from 'path'
export const templateDir = path.resolve(__dirname, '../lib/template/')
export function nvueOutDir() {
  return path.join(process.env.UNI_OUTPUT_DIR, '../.nvue')
}
