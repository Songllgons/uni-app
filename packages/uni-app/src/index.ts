export * from './ssr'
export * from './api'
export * from './hbx'
export * from './utils'
export * from './apiLifecycle'
export * from './vue'
