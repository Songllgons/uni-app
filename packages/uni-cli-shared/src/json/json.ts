import { parse } from 'jsonc-parser'
import { preJson, preUVueJson } from '../preprocess'

export function parseJson(jsonStr: string, shouldPre: boolean = false) {
  return parse(
    shouldPre
      ? process.env.UNI_APP_X === 'true'
        ? preUVueJson(jsonStr)
        : preJson(jsonStr)
      : jsonStr
  )
}
