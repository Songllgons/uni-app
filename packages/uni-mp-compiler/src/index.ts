import { extend } from '@vue/shared'
import {
  type CodegenResult,
  type ParserOptions,
  type RootNode,
  baseParse,
} from '@vue/compiler-core'

import { baseCompile } from './compile'
import { parserOptions } from './parserOptions'
import type { CompilerOptions } from './options'
export type { CompilerOptions } from './options'
export { findProp } from '@vue/compiler-core'
export type {
  DirectiveNode,
  NodeTransform,
  DirectiveTransform,
  TransformContext,
  SimpleExpressionNode,
} from '@vue/compiler-core'

export { genExpr } from './codegen'
export { rewriteExpression } from './transforms/utils'
export { isForElementNode } from './transforms/vFor'
export { transformOn } from './transforms/vOn'
export { transformModel } from './transforms/vModel'
export * from './runtimeHelpers'

export function parse(template: string, options: ParserOptions = {}): RootNode {
  return baseParse(template, extend({}, parserOptions, options))
}

export function compile(
  template: string,
  options: CompilerOptions = {}
): CodegenResult {
  return baseCompile(
    template,
    extend({}, parserOptions, options, {
      directiveTransforms: extend({}, options.directiveTransforms || {}),
    })
  )
}
