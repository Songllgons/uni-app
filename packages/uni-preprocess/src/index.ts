export { preprocess } from './preprocess'
export type {
  PreprocessOptions,
  ProcessContext,
  SourceMapOptions,
} from './preprocess'
