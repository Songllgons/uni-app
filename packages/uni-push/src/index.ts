import GtPush from '../lib/gtpush-min'
import { initPushNotification } from './route'
import { initBroadcastChannel, postPushMessage } from './broadcastChannel'

// if (process.env.UNI_PUSH_DEBUG) {
//   GtPush.setDebugMode(true)
// }

// @ts-expect-error
uni.invokePushCallback({
  type: 'enabled',
})

const appid = process.env.UNI_APP_ID!
if (!appid) {
  Promise.resolve().then(() => {
    // @ts-expect-error
    uni.invokePushCallback({
      type: 'clientId',
      cid: '',
      errMsg: 'manifest.json->appid is required',
    })
  })
} else {
  // #ifdef APP
  initPushNotification()
  // #endif
  // #ifdef H5
  initBroadcastChannel(GtPush)
  // #endif
  // #ifdef MP || APP
  if (typeof uni.onAppShow === 'function') {
    uni.onAppShow(() => {
      GtPush.enableSocket(true)
    })
  }
  // #endif
  GtPush.init({
    appid,
    onError: (res) => {
      console.error(res.error)
      const data = {
        type: 'clientId',
        cid: '',
        errMsg: res.error,
      }
      // @ts-expect-error
      uni.invokePushCallback(data)
      // #ifdef H5
      postPushMessage(data)
      // #endif
    },
    onClientId: (res) => {
      const data = {
        type: 'clientId',
        cid: res.cid,
      }
      // @ts-expect-error
      uni.invokePushCallback(data)
      // #ifdef H5
      postPushMessage(data)
      // #endif
    },
    onlineState: (res) => {
      const data = {
        type: 'lineState',
        online: res.online,
      }
      // @ts-expect-error
      uni.invokePushCallback(data)
      // #ifdef H5
      postPushMessage(data)
      // #endif
    },
    onPushMsg: (res) => {
      const data = {
        type: 'pushMsg',
        message: res.message,
      }
      // @ts-expect-error
      uni.invokePushCallback(data)
      // #ifdef H5
      postPushMessage(data)
      // #endif
    },
  })
  // 仅在 jssdk 中监听
  // #ifdef APP
  uni.onPushMessage((res) => {
    if (
      res.type === 'receive' &&
      res.data &&
      (res.data as any).force_notification
    ) {
      // 创建通知栏
      uni.createPushMessage(res.data)
      // 阻止其他监听器继续监听
      ;(res as any).stopped = true
    }
  })
  // #endif
}
