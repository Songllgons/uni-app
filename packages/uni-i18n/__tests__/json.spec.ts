import { I18N_JSON_DELIMITERS } from '@dcloudio/uni-shared'
import { compileI18nJsonStr, parseI18nJson } from '../src/json'
const delimiters: [string, string] = I18N_JSON_DELIMITERS
describe('parseI18nJson', () => {
  test('pages.json->style', () => {
    const pageMeta = parseI18nJson(
      {
        tabBarIndex: 0,
        bounce: 'vertical',
        navigationBar: {
          titleText: '%component.title%组件',
          buttons: [
            {
              text: '%component.btns.0.text%',
              fontSrc: '/static/uni.ttf',
              fontSize: '22px',
              color: '#FFFFFF',
            },
          ],
        },
        route: 'pages/tabBar/component/component',
      },
      {
        'component.title': '内置',
        'component.btns.0.text': '\ue534',
      },
      delimiters
    ) as Record<string, any>
    expect(pageMeta.navigationBar.titleText).toBe('内置组件')
    expect(pageMeta.navigationBar.buttons[0].text).toBe('\ue534')
  })
})
describe('compileI18nJsonStr', () => {
  test('empty', () => {
    expect(
      compileI18nJsonStr(JSON.stringify({}), {
        locale: '',
        locales: {},
        delimiters,
      })
    ).toBe('{}')
  })
  test('pages.json->tabBar', () => {
    expect(
      compileI18nJsonStr(
        JSON.stringify({
          color: '#7A7E83',
          selectedColor: '#007AFF',
          borderStyle: 'black',
          backgroundColor: '#%tabBar.backgroundColor%',
          list: [
            {
              pagePath: 'pages/tabBar/component/component',
              iconPath: 'static/component.png',
              selectedIconPath: 'static/componentHL.png',
              text: '%tabBar.0.title%',
            },
            {
              pagePath: 'pages/tabBar/API/API',
              iconPath: 'static/api.png',
              selectedIconPath: 'static/apiHL.png',
              text: '%tabBar.1.title%',
            },
          ],
        }),
        {
          locale: 'zh-Hans',
          locales: {
            'zh-Hans': {
              'tabBar.backgroundColor': 'f8f8f8',
              'tabBar.0.title': '组件',
              'tabBar.1.title': '接口',
            },
          },
          delimiters,
        }
      )
    ).toMatchSnapshot()
  })
  test('pages.json->tabBar with multi language', () => {
    expect(
      compileI18nJsonStr(
        JSON.stringify({
          color: '#7A7E83',
          selectedColor: '#007AFF',
          borderStyle: 'black',
          backgroundColor: '#%tabBar.backgroundColor%',
          list: [
            {
              pagePath: 'pages/tabBar/component/component',
              iconPath: 'static/component.png',
              selectedIconPath: 'static/componentHL.png',
              text: '%tabBar.0.title%',
            },
            {
              pagePath: 'pages/tabBar/API/API',
              iconPath: 'static/api.png',
              selectedIconPath: 'static/apiHL.png',
              text: '%tabBar.1.title%',
            },
          ],
        }),
        {
          locale: 'zh-Hans',
          locales: {
            'zh-Hans': {
              'tabBar.backgroundColor': 'f8f8f8',
              'tabBar.0.title': '组件',
              'tabBar.1.title': '接口',
            },
            en: {
              'tabBar.backgroundColor': 'f6f6f6',
              'tabBar.0.title': 'Component',
              'tabBar.1.title': 'API',
            },
          },
          delimiters,
        }
      )
    ).toMatchSnapshot()
  })

  test('androidPrivacy.json', () => {
    expect(
      compileI18nJsonStr(
        JSON.stringify({
          version: '1',
          prompt: 'template',
          title: '%p.title%',
          message: '　　%p.message%',
          buttonAccept: '%p.accept%',
          buttonRefuse: '%p.refuse%',
          second: {
            title: '%p.second.title%',
            message:
              '　　进入应用前，你需先同意<a href="">《服务协议》</a>和<a href="">《隐私政策》</a>，否则将退出应用。',
            buttonAccept: '同意并继续',
            buttonRefuse: '退出应用',
          },
          styles: {
            backgroundColor: '#00FF00',
            borderRadius: '5px',
            title: {
              color: '%p.title.color%',
            },
            buttonAccept: {
              color: '#ffff00',
            },
            buttonRefuse: {
              color: '#00ffff',
            },
          },
        }),
        {
          locale: 'zh-Hans',
          locales: {
            'zh-Hans': {
              'p.title': '服务协议和隐私政策',
              'p.message':
                '请你务必审慎阅读、充分理解“服务协议”和“隐私政策”各条款，包括但不限于：为了更好的向你提供服务，我们需要收集你的设备标识、操作日志等信息用于分析、优化应用性能。<br/>　　你可阅读<a href="">《服务协议》</a>和<a href="">《隐私政策》</a>了解详细信息。如果你同意，请点击下面按钮开始接受我们的服务。',
              'p.accept': '同意并接受',
              'p.refuse': '暂不同意',
              'p.second.title': '确认提示',
              'p.title.color': '#ff00ff',
            },
            en: {
              'p.title': 'Privacy',
              'p.message': 'privacy',
              'p.accept': 'accept',
              'p.refuse': 'refuse',
              'p.second.title': 'confirm',
              'p.title.color': '#ff00ff',
            },
          },
          delimiters,
        }
      )
    ).toMatchSnapshot()
  })
})
