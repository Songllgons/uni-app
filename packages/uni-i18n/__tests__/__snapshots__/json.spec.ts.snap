// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compileI18nJsonStr androidPrivacy.json 1`] = `
"{
  "version": "1",
  "prompt": "template",
  "title": "服务协议和隐私政策",
  "message": "　　请你务必审慎阅读、充分理解“服务协议”和“隐私政策”各条款，包括但不限于：为了更好的向你提供服务，我们需要收集你的设备标识、操作日志等信息用于分析、优化应用性能。<br/>　　你可阅读<a href=\\"\\">《服务协议》</a>和<a href=\\"\\">《隐私政策》</a>了解详细信息。如果你同意，请点击下面按钮开始接受我们的服务。",
  "buttonAccept": "同意并接受",
  "buttonRefuse": "暂不同意",
  "second": {
    "title": "确认提示",
    "message": "　　进入应用前，你需先同意<a href=\\"\\">《服务协议》</a>和<a href=\\"\\">《隐私政策》</a>，否则将退出应用。",
    "buttonAccept": "同意并继续",
    "buttonRefuse": "退出应用",
    "titleLocales": {
      "zh-Hans": "确认提示",
      "en": "confirm"
    }
  },
  "styles": {
    "backgroundColor": "#00FF00",
    "borderRadius": "5px",
    "title": {
      "color": "#ff00ff",
      "colorLocales": {
        "zh-Hans": "#ff00ff",
        "en": "#ff00ff"
      }
    },
    "buttonAccept": {
      "color": "#ffff00"
    },
    "buttonRefuse": {
      "color": "#00ffff"
    }
  },
  "titleLocales": {
    "zh-Hans": "服务协议和隐私政策",
    "en": "Privacy"
  },
  "messageLocales": {
    "zh-Hans": "　　请你务必审慎阅读、充分理解“服务协议”和“隐私政策”各条款，包括但不限于：为了更好的向你提供服务，我们需要收集你的设备标识、操作日志等信息用于分析、优化应用性能。<br/>　　你可阅读<a href=\\"\\">《服务协议》</a>和<a href=\\"\\">《隐私政策》</a>了解详细信息。如果你同意，请点击下面按钮开始接受我们的服务。",
    "en": "　　privacy"
  },
  "buttonAcceptLocales": {
    "zh-Hans": "同意并接受",
    "en": "accept"
  },
  "buttonRefuseLocales": {
    "zh-Hans": "暂不同意",
    "en": "refuse"
  }
}"
`;

exports[`compileI18nJsonStr pages.json->tabBar 1`] = `
"{
  "color": "#7A7E83",
  "selectedColor": "#007AFF",
  "borderStyle": "black",
  "backgroundColor": "#f8f8f8",
  "list": [
    {
      "pagePath": "pages/tabBar/component/component",
      "iconPath": "static/component.png",
      "selectedIconPath": "static/componentHL.png",
      "text": "组件"
    },
    {
      "pagePath": "pages/tabBar/API/API",
      "iconPath": "static/api.png",
      "selectedIconPath": "static/apiHL.png",
      "text": "接口"
    }
  ]
}"
`;

exports[`compileI18nJsonStr pages.json->tabBar with multi language 1`] = `
"{
  "color": "#7A7E83",
  "selectedColor": "#007AFF",
  "borderStyle": "black",
  "backgroundColor": "#f8f8f8",
  "list": [
    {
      "pagePath": "pages/tabBar/component/component",
      "iconPath": "static/component.png",
      "selectedIconPath": "static/componentHL.png",
      "text": "组件",
      "textLocales": {
        "zh-Hans": "组件",
        "en": "Component"
      }
    },
    {
      "pagePath": "pages/tabBar/API/API",
      "iconPath": "static/api.png",
      "selectedIconPath": "static/apiHL.png",
      "text": "接口",
      "textLocales": {
        "zh-Hans": "接口",
        "en": "API"
      }
    }
  ],
  "backgroundColorLocales": {
    "zh-Hans": "#f8f8f8",
    "en": "#f6f6f6"
  }
}"
`;
